Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUExternalTextureDescriptor.h

```diff
@@ -33,24 +33,20 @@
 typedef struct __CVBuffer* CVPixelBufferRef;
 #endif
 
-namespace PAL::WebGPU {
+namespace WebCore {
+class VideoFrame;
```

Comment:
We are in PAL, so we cannot really refer to VideoFrame.
We could probably move VideoFrame to PAL.

Reply:
Agreed this is probably a layering violation which only works as this is a forward declaration. I will see how much effort it would be to move VideoFrame to PAL.

Reply:
@youennf I updated the PR with creating a PAL::VideoFrame and added some static downcasts. I used `static_cast` as all`PAL::VideoFrame` instances are `WebCore::` ones to avoid any performance loss with `dynamicDowncast`. If you think there are any issues with this idea please let me know.

I initially attempted to move all of `VideoFrame` to PAL but this is a very large change as it pulls in several WebCore types. I think it is far too risky given where we are in the software development cycle.